### PR TITLE
[WINMM] Load Wave/MIDI mapper drivers only after actual device drivers

### PR DESCRIPTION
**Pull request by binarymaster**
## Purpose

Load Wave/MIDI mapper drivers only after actual device drivers.

JIRA issue: [CORE-15620](https://jira.reactos.org/browse/CORE-15620)

**Note:** winmm is not synced with Wine (it's forked), so I did not used `__REACTOS__` ifdefs.

### CODE REVIEW

File: dll/win32/winmm/registry.c [hunk 1]
```diff
@@ -60,14 +63,16 @@ BOOL LoadRegistryMMEDrivers(char* key)
             if ( ! stricmp("wavemapper", value_name) )
             {
                 TRACE("Found a Wave-mapper: %s\n", value_data);
-                valid_driver = TRUE;
+                /* Delay loading Wave mapper driver */
+                strcpy(wavemapper, value_data);
```
Comment (learn-more) [inline dll/win32/winmm/registry.c]: What if there are 2 wave mapper drivers, is this possible?

Comment (binarymaster) [reply dll/win32/winmm/registry.c]: Not possible, there is only one mapper for wave and one for MIDI.

Each mapper forwards the streams to one or many actual devices depending on rules in registry.

**Comment by yagoulas** [comment]
I was wondering, wine has some other code to load drivers, would some of that be useful to us?

**Comment by binarymaster** [comment]
> I was wondering, wine has some other code to load drivers, would some of that be useful to us?

https://source.winehq.org/git/wine.git/blob/8e0139af2a1cd1ac12aadfa18c8bb04552433ce4:/dlls/winmm/lolvldrv.c#l491

From what I see here, they also modified winmm to load mappers after all device drivers, but they hardcoded them. Maybe because they changed a way of drivers enumeration (registry vs. IMMDeviceEnumerator).

Not sure if this is useful to us.

**Pull request merged**